skrifa/src/outline/embedded_hinting.rs

```diff
@@ -5,23 +5,68 @@ use super::{
     OutlineGlyph, OutlineGlyphCollection, OutlineKind, OutlinePen, Size,
 };
 
-/// Modes for embedded hinting.
+/// Modes that control hinting when using embedded instructions.
 ///
-/// Only the `glyf` format supports all hinting modes.
+/// Only the TrueType interpreter supports all hinting modes.
+///
+/// # FreeType compatibility
+///
+/// The following table describes how to map FreeType hinting modes:
+///
+/// | FreeType mode         | Variant                                                                              |
+/// |-----------------------|--------------------------------------------------------------------------------------|
+/// | FT_LOAD_TARGET_MONO   | Strong                                                                               |
+/// | FT_LOAD_TARGET_NORMAL | Smooth { lcd_subpixel: None, preserve_linear_metrics: false }                        |
+/// | FT_LOAD_TARGET_LCD    | Smooth { lcd_subpixel: Some(LcdLayout::Horizontal), preserve_linear_metrics: false } |
+/// | FT_LOAD_TARGET_LCD_V  | Smooth { lcd_subpixel: Some(LcdLayout::Vertical), preserve_linear_metrics: false }   |
+///
+/// Note: `FT_LOAD_TARGET_LIGHT` is equivalent to `FT_LOAD_TARGET_NORMAL` since
+/// FreeType 2.7.
+///
+/// The default value of this type is equivalent to `FT_LOAD_TARGET_NORMAL`.
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub enum HintingMode {
+    /// Strong hinting mode that should only be used for aliased, monochromatic
+    /// rasterization.
+    ///
+    /// Corresponds to `FT_LOAD_TARGET_MONO` in FreeType.
+    Strong,
+    /// Lighter hinting mode that is intended for anti-aliased rasterization.
+    Smooth {
+        /// If set, enables support for optimized hinting that takes advantage
+        /// of subpixel layouts in LCD displays and corresponds to
+        /// `FT_LOAD_TARGET_LCD` or `FT_LOAD_TARGET_LCD_V` in FreeType.
+        ///
+        /// If unset, corresponds to `FT_LOAD_TARGET_NORMAL` in FreeType.
+        lcd_subpixel: Option<LcdLayout>,
+        /// If true, prevents adjustment of the outline in the horizontal
+        /// direction and preserves inter-glyph spacing.
+        ///
+        /// This has no corresponding setting in FreeType.
```

Comment:
optional, indicating when you should use it might be nice

Reply:
Done